share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorService.java

```diff
@@ -91,6 +93,8 @@ public class ShareCoordinatorService implements ShareCoordinator {
     private final Timer timer;
     private final PartitionWriter writer;
     private final Map<TopicPartition, Long> lastPrunedOffsets;
+    private Supplier<Boolean> isShareGroupConfigEnabled = () -> true;
```

Comment:
Shouldn't this be `final` and not have default supplier as there is a check in builder that it cannot be null?
```suggestion
    private final Supplier<Boolean> isShareGroupConfigEnabled;
```